packages/stream_chat/lib/src/models/filter.dart

```diff
@@ -0,0 +1,197 @@
+// ignore_for_file: non_constant_identifier_names, constant_identifier_names
+
+const _groupOperators = [
+  FilterOperator.and,
+  FilterOperator.or,
+  FilterOperator.nor,
+];
+
+/// Possible operators to use in filters.
+enum FilterOperator {
+  /// Matches values that are equal to a specified value.
+  equal,
+
+  /// Matches all values that are not equal to a specified value.
+  notEqual,
+
+  /// Matches values that are greater than a specified value.
+  greater,
+
+  /// Matches values that are greater than a specified value.
+  greaterOrEqual,
+
+  /// Matches values that are less than a specified value.
+  less,
+
+  /// Matches values that are less than or equal to a specified value.
+  lessOrEqual,
+
+  /// Matches any of the values specified in an array.
+  in_,
+
+  /// Matches none of the values specified in an array.
+  notIn,
+
+  /// Matches values by performing text search with the specified value.
+  query,
+
+  /// Matches values with the specified prefix.
+  autoComplete,
+
+  /// Matches values that exist/don't exist based on the specified boolean value.
+  exists,
+
+  /// Matches all the values specified in an array.
+  and,
+
+  /// Matches at least one of the values specified in an array.
+  or,
+
+  /// Matches none of the values specified in an array.
+  nor,
+}
+
+/// Helper extension for [FilterOperator]
+extension FilterOperatorX on FilterOperator {
+  /// Converts [FilterOperator] into rew values
+  String get rawValue => {
+        FilterOperator.equal: '\$eq',
+        FilterOperator.notEqual: '\$ne',
+        FilterOperator.greater: '\$gt',
+        FilterOperator.greaterOrEqual: '\$gte',
+        FilterOperator.less: '\$lt',
+        FilterOperator.lessOrEqual: '\$lte',
+        FilterOperator.in_: '\$in',
+        FilterOperator.notIn: '\$nin',
+        FilterOperator.query: '\$q',
+        FilterOperator.autoComplete: '\$autocomplete',
+        FilterOperator.exists: '\$exists',
+        FilterOperator.and: '\$and',
+        FilterOperator.or: '\$or',
+        FilterOperator.nor: '\$nor',
+      }[this]!;
+}
+
+/// Stream supports a limited set of filters for querying channels,
+/// users and members. The example below shows how to filter for channels
+/// of type messaging where the current user is a member
+///
+/// ```dart
+/// val filter = Filter.and(
+///     Filter.equal('type', 'messaging'),
+///     Filter.in_('members', [user.id])
+/// )
+/// ```
+/// See <a href="https://getstream.io/chat/docs/query_channels/?language=dart" target="_top">Query Channels Documentation</a>
+class Filter {
+  const Filter.__({
```

Comment:
what do you think about giving a name to these internal constructors?

Reply:
Any particular use-case for that? I don't really think we need to name them.

Reply:
we don't actually need to, it was just for readability